src/lunaCatcher.ts

```diff
@@ -1,157 +1,184 @@
+import * as ks from "./keystore";
+
+const ONE_LUNA = 10 ^ 6;
+
+async function initKey(
+  keyPath: string,
+  keyName: string,
+  password?: string
+): Promise<RawKey> {
+  const plainEntity = ks.load(
+    keyPath,
+    keyName,
+    password ||
+      (await promptly.password(`Enter a passphrase:`, { replace: `*` }))
+  );
+
+  return new RawKey(Buffer.from(plainEntity.privateKey, "hex"));
 }
 
-
 // yarn start catch command
 export async function processCatch(
-    client: LCDClient,
-    wallet: Wallet,
-    sourceAccount: string,
-    targetAccount: string
+  client: LCDClient,
+  wallet: Wallet,
+  sourceAccount: string,
+  targetAccount: string,
+  feeLuna: number
 ): Promise<void> {
-    
-    // Print timestamp before start
-    console.info(`${new Date().toUTCString()}, checking account balance...`);
-
-    const balance = await client.bank.balance(sourceAccount);
-    const ulunaBalance = balance[0].get("uluna");
-    
-    if (ulunaBalance) {
-
-        var accountBalanceAsNumber = Number(ulunaBalance.amount);
-
-        console.dir(`Current balance: ${accountBalanceAsNumber}uluna `);
-
-        if (accountBalanceAsNumber >= 500000) {
-
-            console.dir(`Balance is higher or equal to 1 Luna, trying to transfer to target account...`);
-
-            const fee = 100000;
-            const amountToTransfer = accountBalanceAsNumber - fee;
-
-            console.dir(`Amount to transfer ${amountToTransfer}uluna`);
-
-            // create a simple message that moves coin balances
-            const send = new MsgSend(
-                sourceAccount,
-                targetAccount,
-                { uluna: amountToTransfer }
-            );
-
-            // sign and send
-            const msgs = [send];
-            const tx = await wallet.createAndSignTx({
-                msgs,
-                fee: new Fee(100000, [new Coin("uluna", fee)]),
-                memo: `Don't be evil. Funds rescue powered by Luna Station 88.`,
-            });
-
-            const res = await client.tx.broadcastSync(tx).catch((err) => {
-                console.error(`broadcast error: ${err.message}`, tx.toData());
-                throw err;
-            });
-
-            if (isTxError(res)) {
-                console.error(`broadcast error: code: ${res.code}, raw_log: ${res.raw_log}`);
-                return;
-            }
-
-            const txhash = res.txhash;
-            console.info(`broadcast success: txhash: ${txhash}`);
-
-            await Bluebird.delay(3000);
-        }
+  // Print timestamp before start
+  console.info(`${new Date().toUTCString()}, checking account balance...`);
+
+  const balance = await client.bank.balance(sourceAccount);
+  const ulunaBalance = balance[0].get("uluna");
+
+  if (ulunaBalance) {
+    var accountBalanceAsNumber = Number(ulunaBalance.amount);
+
+    console.dir(`Current balance: ${accountBalanceAsNumber}uluna `);
+
+    if (accountBalanceAsNumber >= (feeLuna + 0.5) * ONE_LUNA) {
```

Comment:
keep the threshold slightly above the fee to avoid attempting to transfer less than the fee.